featureflags.go

```diff
@@ -242,8 +220,23 @@ func (poller *FeatureFlagsPoller) GetFeatureFlags() []FeatureFlag {
 
 func (poller *FeatureFlagsPoller) request(method string, endpoint string, requestData []byte, headers [][2]string) (*http.Response, error) {
 
-	url := poller.Endpoint + "/" + endpoint + "/?token=" + poller.projectApiKey + ""
-	req, err := http.NewRequest(method, url, bytes.NewReader(requestData))
+	url, err := url.Parse(poller.Endpoint + "/" + endpoint + "")
+
+	if err != nil {
+		poller.Errorf("creating url - %s", err)
+	}
+	searchParams := url.Query()
+
+	if method == "GET" {
+		searchParams.Add("token", poller.projectApiKey)
+	}
+
+	if endpoint == "decide" {
+		searchParams.Add("v", "2")
```

Comment:
this feels too obfuscated, since there's no `decide` anymore, only `decide?v=2`. Thoughts on making either searchParams a parameter, or updating endpoint to be `decide?v=2` . Either should work with `url.Parse()`